src/loader.ts

```diff
@@ -1,207 +1,101 @@
-    const moduleCachePath = await parseModule(specifier, modulePath);
-    return nextResolve(moduleCachePath);
-};
-
-/**
- * parseModule
- * @description a convenience function to parse the modules, this function is called by the loader automatically
- * @param specifier string
- * @param modulePath string
- * @returns string
- */
-export const parseModule = async (specifier: string, modulePath: string) => {
-  const { cache } = constructLoaderConfig()
-  const { protocol } = new URL(modulePath);
-  const isNode = protocol === "node:";
-  const isFile = protocol === "file:";
-
-  if (isNode || isFile) return specifier;
-
-  const cachePath = constructCachePath({ cache, modulePath })
-
-  cacheMap.set(`file://${cachePath}`, modulePath);
-  if (existsSync(cachePath)) return cachePath
-  const code = await (await fetch(modulePath).then(response => {
-    if (!response.ok) {
-      throw Error(`404: Module not found: ${modulePath}`);
-    }
-    return response;
-  })).text();
-
-  ensureFileSync(cachePath);
-  writeFileSync(cachePath, code);
-
-  return cachePath;
+export const resolve = async (
+  specifier: string,
+  { parentURL }: Context,
+  nextResolve: NextResolve,
+  options: ResolveOptions = config?.values,
+) => {
+  const { basePath, cachePath, debug: isDebugging = false, importmapPath, cacheMap = initialCacheMap } = options || {};
+  log.setLogger(isDebugging);
+  try {
+    // define importmap path
+    const cwd = process.cwd();
+    const pathToImportMap = importmapPath || constructUrlPath(basePath, cwd, isDebugging);
+    const nodeImportMapPath = constructPath("node.importmap", pathToImportMap);
+    log.debug("resolve:nodeImportMapPath:", { cwd, pathToImportMap, nodeImportMapPath });
+    if (!nodeImportMapPath) throw new Error("Failed in resolving import map path");
+
+    // define cache path
+    const pathToCache = cachePath || parentURL;
+    log.debug("resolve:pathToCache:", { pathToCache });
+    if (!pathToCache) throw new Error("Failed in resolving cache path");
+
+    // construct importmap
+    const importmap = constructImportMap(nodeImportMapPath);
+    log.debug("resolve:importmap:", { importmap });
+    if (!importmap) throw new Error("Failed in constructing import map");
+
+    // construct cache map path
+    const cacheMapPath = cacheMap.get(pathToCache);
+    log.debug("resolve:cacheMapPath:", { cacheMapPath });
+    if (!cacheMapPath) throw new Error("Failed in resolving cache map path");
+
+    // construct module path
+    const modulePath = importmap.resolve(specifier, cacheMapPath);
+    log.debug("resolve:modulePath:", { modulePath });
+    if (!modulePath) throw new Error("Failed in resolving module path");
+
+    // resolve URL
+    const { protocol = "" } = new URL(modulePath);
+    const isNode = protocol === "node:";
+    const isFile = protocol === "file:";
+    log.debug("resolve:protocol:", { protocol, isNode, isFile });
+    if (isNode || isFile) throw new Error("Failed in resolving URL");
```

Comment:
If we detect these protocols. We shouldn't throw error. Because these are supported natively in `node` and these are node. We just need to do this
`return nextResolve(specifier`. 

Eg: if this is `node:process`. It loads from the native `node` process. Instead of throwing error.